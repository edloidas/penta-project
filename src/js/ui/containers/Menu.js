// @flow
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import MainMenu from '../components/MainMenu/MainMenu';
import { type MainMenuItemProps } from '../components/MainMenu/MainMenuItem';
import * as GameActions from '../actions/game';

type Props = {
  actions: {
    resumeGame: typeof GameActions.resumeGame
  }
};

// eslint-disable-next-line react/prefer-stateless-function
class Menu extends Component<Props> {
  constructor(props) {
    super(props);
    this.handleResume = this.handleResume.bind(this);
  }

  props: Props;

  handleResume: (e?: MouseEvent) => void;

  handleResume(e: MouseEvent) {
    e.stopPropagation();
    e.preventDefault();
    this.props.actions.resumeGame(true);
  }

  render() {
    const menuItems: Array<MainMenuItemProps> = [
      { name: 'Resume', clickHandler: this.handleResume },
      { name: 'New Game', to: '/new' },
      { name: 'Save Game', to: '/save' },
      { name: 'Load Game', to: '/load' },
      { name: 'Hero', to: '/hero' },
      { name: 'Settings', to: '/settings' },
      { name: 'Exit', to: '/exit' }
    ];
    return <MainMenu menuItems={menuItems} />;
  }
}

function mapStateToProps() {
  return {};
}

function mapDispatchToProps(dispatch: Function) {
  return {
    actions: bindActionCreators(GameActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
