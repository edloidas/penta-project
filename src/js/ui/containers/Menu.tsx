import * as React from 'react';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import MainMenu from '../components/MainMenu/MainMenu';
import { MainMenuItemProps } from '../components/MainMenu/MainMenuItem';
import * as GameActions from '../actions/game';
import Focusable from './Focusable';

type Props = {
  actions: {
    resumeGame: typeof GameActions.resumeGame,
    startNewGame: typeof GameActions.startNewGame,
    saveGame: typeof GameActions.saveGame,
    loadGame: typeof GameActions.loadGame
  }
};

// eslint-disable-next-line react/prefer-stateless-function
class Menu extends Focusable<Props> {
  constructor(props) {
    super(props);
    this.handleResume = this.handleResume.bind(this);

    this.menuItems = [
      { name: 'Resume', clickHandler: this.handleResume },
      { name: 'New Game', to: '/new' },
      { name: 'Save Game', to: '/save' },
      { name: 'Load Game', to: '/load' },
      { name: 'Hero', to: '/hero' },
      { name: 'Settings', to: '/settings' },
      { name: 'Exit', to: '/exit' }
    ];
  }

  props: Props;

  menuItems: Array<MainMenuItemProps>;

  handleResume(e: MouseKeyboardEvent) {
    e.stopPropagation();
    e.preventDefault();
    this.props.actions.resumeGame(true);
  }

  render() {
    return (
      <div
        tabIndex={0}
        role="button"
        ref={div => {
          this.focusable = div;
        }}>
        <MainMenu menuItems={this.menuItems} />
      </div>
    );
  }
}

function mapStateToProps() {
  return {};
}

function mapDispatchToProps(dispatch: Dispatch<GameActions.ActionsTypes>) {
  return {
    actions: bindActionCreators(GameActions, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Menu);
