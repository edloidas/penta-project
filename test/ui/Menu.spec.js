import React from 'react';
import { Provider } from 'react-redux';
import { ConnectedRouter as Router } from 'react-router-redux';
import renderer from 'react-test-renderer';
import configureStore from '../../src/js/ui/store';
import history from '../../src/js/ui/store/history';
import Menu from '../../src/js/ui/containers/Menu';

describe('<Menu />', () => {
  test('Should renders', () => {
    const store = configureStore();

    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history}>
            <Menu />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
