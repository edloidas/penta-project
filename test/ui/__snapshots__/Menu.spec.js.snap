// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Should renders 1`] = `
<div
  role="button"
  tabIndex="0"
>
  <div
    className="effect__appear sc-bwzfXH iVThND"
  >
    <div
      className="sc-gZMcBi qaFpZ"
    >
      <div
        className="sc-gqjmRU cBrrLK"
      >
        <div
          className="font__mono sc-htpNat easqzJ"
        >
          <div
            className="sc-ifAKCX RfDSL"
          >
            Penta Project
          </div>
          <div
            className="sc-EHOje bMrtUk"
          >
            A Cyberpunk Action Game
          </div>
        </div>
      </div>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="Resume"
        onClick={[Function]}
        tabIndex={0}
      >
        Resume
      </button>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="New Game"
        onClick={null}
        tabIndex={-1}
      >
        New Game
        <a
          href="/new"
          onClick={[Function]}
          tabIndex="0"
        />
      </button>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="Save Game"
        onClick={null}
        tabIndex={-1}
      >
        Save Game
        <a
          href="/save"
          onClick={[Function]}
          tabIndex="0"
        />
      </button>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="Load Game"
        onClick={null}
        tabIndex={-1}
      >
        Load Game
        <a
          href="/load"
          onClick={[Function]}
          tabIndex="0"
        />
      </button>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="Hero"
        onClick={null}
        tabIndex={-1}
      >
        Hero
        <a
          href="/hero"
          onClick={[Function]}
          tabIndex="0"
        />
      </button>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="Settings"
        onClick={null}
        tabIndex={-1}
      >
        Settings
        <a
          href="/settings"
          onClick={[Function]}
          tabIndex="0"
        />
      </button>
      <button
        className="effect__glitch sc-iwsKbI ixchWs"
        data-text="Exit"
        onClick={null}
        tabIndex={-1}
      >
        Exit
        <a
          href="/exit"
          onClick={[Function]}
          tabIndex="0"
        />
      </button>
    </div>
    <footer
      className="font__accent sc-VigVT byUSmW"
    >
      <div
        className="sc-jTzLTM iHsIJR"
      >
        <span
          className="sc-bZQynM fjiOsu"
        >
          <span
            className="sc-gzVnrw hjuePL"
          >
            alpha version
          </span>
           
          "0.0.1-alpha.2"
        </span>
      </div>
      <span
        className="sc-htoDjs dNFyTW"
      >
        2017–current © 
        edloidas production
        . 
        <span
          className="sc-dnqmqq oqlux"
        >
          All Rights Reserved.
        </span>
      </span>
    </footer>
  </div>
</div>
`;
